cmd/swarm/swarm-smoke/upload_and_sync.go

```diff
@@ -58,6 +58,21 @@ func cliUploadAndSync(c *cli.Context) error {
 	log.PrintOrigins(true)
 	log.Root().SetHandler(log.LvlFilterHandler(log.Lvl(verbosity), log.StreamHandler(os.Stdout, log.TerminalFormat(true))))
 
+	errc := make(chan error)
+	go func() {
+		errc <- uploadAndSync(c)
+	}()
+
+	select {
+	case err := <-errc:
+		return err
+	case <-time.After(time.Duration(timeout) * time.Second):
```

Comment:
Actually I am not sure if K8s won't reschedule a failed cron job immediately. It could be the case that we must always return an exit code of 0 to the OS, even in the case of failure, and only record failures as metrics.